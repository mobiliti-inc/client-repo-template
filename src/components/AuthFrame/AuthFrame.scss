@import '~variables';

.auth-frame {
	display: flex;
	width: 100%;
	height: 100vh;
	@media (max-width: $mobile-width) {
		justify-content: center;
	}
}

.section-auth-content {
	min-width: 40%;
	height: 100%;
	overflow-y: scroll;
	padding: 0 5%;

	@media (max-width: $mobile-width) {
		display: flex;
		flex-direction: column;
		margin-top: 30px;
		padding: 0 10%;
		min-width: 100%;
	}
}

.navigation-container {
	display: flex;
	flex-direction: row;
	justify-content: space-between;
}

.back-arrow-container {
	display: none;
	height: 18px;
	width: 24px;
	justify-self: flex-start;

	@media (max-width: $mobile-width) {
		display: block;
	}
}

.back-arrow-image {
	height: 100%;
	width: 100%;
	cursor: pointer;
}

.close-icon-container {
	display: none;
	height: 18px;
	width: 24px;
	margin-left: auto;

	@media (max-width: $mobile-width) {
		display: block;
	}
}

.close-icon {
	height: 100%;
	width: 100%;
}

.brand-logo-container {
	padding-top: 75px;
	display: block;
	@media (max-width: $mobile-width) {
		display: none;
	}
}

.brand-logo {
	height: 22px;
	width: 116px;
}

.section-cover-image {
	width: 60%;
	min-height: 100%;
	overflow: hidden;

	@media (max-width: $mobile-width) {
		display: none;
	}
}

.section-cover-image img {
	width: 100%;
	height: 100%;
	object-fit: cover;
}

.main-content {
	min-width: 10px;
}

.mobile-store-container {
	height: 38px;
	cursor: pointer;
	margin-top: auto;
	display: block;
	margin-bottom: 50px;
	@media (max-width: $mobile-width) {
		display: none;
	}
}

.img-app-store {
	width: 128px;
	height: 100%;
	margin-right: 10px;
}

.img-play-store {
	width: 128px;
	height: 100%;
}
