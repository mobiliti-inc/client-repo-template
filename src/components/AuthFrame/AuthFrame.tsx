import React, { PureComponent } from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router";
import styles from "./AuthFrame.scss";
import brandLogo from "../../assets/icons/logo_Dark.svg";
import appStoreBadge from "../../assets/icons/app-store.svg";
import googleStoreBadge from "../../assets/icons/google-play.svg";
import backArrowImage from "../../assets/icons/icon_Arrow-Left.svg";
import imgSignupBackground from "../../assets/images/img_SignIn-Background.png";
import { iconClose } from "../../assets";
import Link from "../Link/Link";

type AuthFrameProps = {
	showBackButton?: boolean,
	showCloseButton?: boolean,
	history: {
		goBack: (...args: any[]) => any
	},
	userIsLoggedIn: boolean
};

class AuthFrame extends PureComponent<AuthFrameProps, {}> {
	componentWillMount = () => {
		const { history, userIsLoggedIn } = this.props;
		if (userIsLoggedIn) {
			history.goBack();
		}
	}

	render() {
		const { children, showBackButton, showCloseButton } = this.props;
		return (
			<div styleName="auth-frame">
				<div styleName="section-auth-content">
					<div styleName="navigation-container">
						{showBackButton ? (
							<div
								styleName="back-arrow-container"
								onClick={this.props.history.goBack}
								role="presentation"
							>
								<img src={backArrowImage} styleName="back-arrow-image" alt="back arrow" />
							</div>
						) : null}
						{showCloseButton ? (
							<div styleName="close-icon-container">
								<img src={iconClose} styleName="close-icon" alt="close" />
							</div>
						) : null}
					</div>
					<div styleName="brand-logo-container">
						<img styleName="brand-logo" src={brandLogo} alt="brand-logo" />
					</div>
					<div styleName="main-content">{children}</div>
					<div styleName="mobile-store-container">
						<Link to="https://itunes.apple.com/us/app/mobiliti/id1326755398?ls=1&amp;mt=8" external>
							<img styleName="img-app-store" src={appStoreBadge} alt="app-store" />
						</Link>
						<Link to="https://play.google.com/store/apps/details?id=com.mobiliti" external>
							<img styleName="img-play-store" src={googleStoreBadge} alt="play-store" />
						</Link>
					</div>
				</div>
				<div styleName="section-cover-image">
					<img src={imgSignupBackground} alt="signup background" />
				</div>
			</div>
		);
	}
}

const mapStateToProps = state => ({
	userIsLoggedIn: state.auth.isLoggedIn
});

export const AuthFrameComponent = CSSModules(AuthFrame, styles, { allowMultiple: true });

export default withRouter(connect(mapStateToProps)(AuthFrameComponent));
