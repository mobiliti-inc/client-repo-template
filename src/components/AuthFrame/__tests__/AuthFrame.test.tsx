import { AuthFrameComponent } from '../AuthFrame';

describe('AuthFrame', () => {
	const initialProps = {
		history: {
			goBack: jest.fn(),
		},
		userIsLoggedIn: false,
	};
	it('renders', () => {
		const subject = shallow(
			<AuthFrameComponent
				{...initialProps}
			>Hello world!
			</AuthFrameComponent>);
		matchSnapshot(subject);
	});

	it('renders the back button by default', () => {
		const subject = shallow(
			<AuthFrameComponent
				{...initialProps}
			>Hello world!
			</AuthFrameComponent>);
		expect(subject.find('.back-arrow-container')).toHaveLength(1);
	});

	it('does not renders the back button when showBackButton is false', () => {
		const subject = shallow(
			<AuthFrameComponent {...initialProps} showBackButton={false}>Hello world!</AuthFrameComponent>
		);

		expect(subject.find('.back-arrow-container')).toHaveLength(0);
	});
});
