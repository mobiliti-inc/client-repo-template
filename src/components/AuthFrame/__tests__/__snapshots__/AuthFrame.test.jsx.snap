// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthFrame renders 1`] = `
<div
  className="auth-frame"
>
  <div
    className="section-auth-content"
  >
    <div
      className="navigation-container"
    >
      <div
        className="back-arrow-container"
        onClick={[MockFunction]}
        role="presentation"
      >
        <img
          alt="back arrow"
          className="back-arrow-image"
          src="test-file-stub"
        />
      </div>
    </div>
    <div
      className="brand-logo-container"
    >
      <img
        alt="brand-logo"
        className="brand-logo"
        src="test-file-stub"
      />
    </div>
    <div
      className="main-content"
    >
      Hello world!
    </div>
    <div
      className="mobile-store-container"
    >
      <StyledLink
        alternate={false}
        buttonStyle={false}
        external={true}
        to="https://itunes.apple.com/us/app/mobiliti/id1326755398?ls=1&mt=8"
      >
        <img
          alt="app-store"
          className="img-app-store"
          src="test-file-stub"
        />
      </StyledLink>
      <StyledLink
        alternate={false}
        buttonStyle={false}
        external={true}
        to="https://play.google.com/store/apps/details?id=com.mobiliti"
      >
        <img
          alt="play-store"
          className="img-play-store"
          src="test-file-stub"
        />
      </StyledLink>
    </div>
  </div>
  <div
    className="section-cover-image"
  >
    <img
      alt="signup background"
      src="test-file-stub"
    />
  </div>
</div>
`;
